### PR TITLE
Make sure to disallow creating a replicated distributed table concurrently

### PR DESCRIPTION
**Pull request by onurctirtir**
See explanation in https://github.com/citusdata/citus/issues/7216.
Fixes https://github.com/citusdata/citus/issues/7216.

DESCRIPTION: Makes sure to disallow creating a replicated distributed table concurrently

### CODE REVIEW

File: src/backend/distributed/commands/create_distributed_table.c [hunk 1]
```diff
@@ -436,6 +436,11 @@ CreateDistributedTableConcurrently(Oid relationId, char *distributionColumnName,
 
 	char replicationModel = DecideDistTableReplicationModel(distributionMethod,
 															colocateWithTableName);
+	if (replicationModel == REPLICATION_MODEL_COORDINATOR)
```
Comment (onderkalaci) [inline src/backend/distributed/commands/create_distributed_table.c:439]: maybe != REPLICATION_MODEL_STREAMING is safer?

File: src/backend/distributed/commands/create_distributed_table.c [hunk 2]
```diff
@@ -436,6 +436,11 @@ CreateDistributedTableConcurrently(Oid relationId, char *distributionColumnName,
...
+	{
+		ereport(ERROR, (errmsg("cannot create a replicated distributed a "
```
Comment (onurctirtir) [inline src/backend/distributed/commands/create_distributed_table.c:441]:
```suggestion
		ereport(ERROR, (errmsg("cannot create a replicated distributed "
```

File: src/test/regress/expected/create_distributed_table_concurrently.out [hunk 3]
```diff
@@ -36,6 +36,18 @@ set citus.shard_replication_factor to 2;
...
+select create_distributed_table('dist_1', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+set citus.shard_replication_factor to 1;
+create table dist_2(a int);
+select create_distributed_table_concurrently('dist_2', 'a', colocate_with=>'dist_1');
+ERROR:  cannot create a replicated distributed a table concurrently
```
Comment (onurctirtir) [inline src/test/regress/expected/create_distributed_table_concurrently.out:50]:
```suggestion
ERROR:  cannot create a replicated distributed table concurrently
```

File: src/backend/distributed/commands/create_distributed_table.c [hunk 4]
```diff
@@ -436,6 +436,11 @@ CreateDistributedTableConcurrently(Oid relationId, char *distributionColumnName,
 
 	char replicationModel = DecideDistTableReplicationModel(distributionMethod,
 															colocateWithTableName);
+	if (replicationModel != REPLICATION_MODEL_STREAMING)
+	{
+		ereport(ERROR, (errmsg("cannot create a replicated distributed "
```
Comment (onderkalaci) [inline src/backend/distributed/commands/create_distributed_table.c:441]: I think we don't use `replicated distributed table` terminology much.

Maybe: errmsg: Citus only allows concurrent table distribution for distributed tables when citus.shard_replication_factor > 1 
             errhint: table %s is colocated with %s which has citus.shard_replication_factor > 1

and, before giving the hint, we could assert for !IsColocateWithDefault/!IsColocateWithNone. In case these are true, we'd already give this error above with ShardReplicationFactor > 1

**Review by onderkalaci — APPROVED** [review]
minor error message suggestion. I think without a hint, it might be hard to follow this is caused by colocation
